### DISCUSSION TITLE
help with a converter

### DISCUSSION THREAD

**Discussion by bogorad** [manna-harbour/miryoku#322] [Q&A] [answered]
I ordered and got this Adafruit rp2040 with USB-A host, planned to use it between my Kinesis KB500 and PC. Here's the repo:

https://github.com/whyaaronbailey/adafruit_rp2040_usbh

Miryoku is installed and successfully compiles for other keyboards. 

```
❯ qmk compile -j 7 -kb beekeeb/piantor_pro -km manna-harbour_miryoku
Ψ Compiling keymap with gmake --jobs=7 --output-sync=target beekeeb/piantor_pro:manna-harbour_miryoku


QMK Firmware 0.20.6
Making beekeeb/piantor_pro with keymap manna-harbour_miryoku

arm-none-eabi-gcc (15:10.3-2021.07-4) 10.3.1 20210621 (release)
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Copying beekeeb_piantor_pro_manna-harbour_miryoku.uf2 to qmk_firmware folder                        [OK]
(Firmware size check does not yet support RP2040; skipping)
```

Tried it - it compiles with QMK using the provided defaults but fails to recognize Miryoku as a valid layout. What am I missing? Must be something in the info.json? 

```
❯ qmk compile -j 7 -kb converter/adafruit_rp2040_usbh -km manna-harbour_miryoku
☒ Invalid keymap argument.
usage: qmk compile [-h] [-c] [-e ENV] [-j PARALLEL] [-n] [-km KEYMAP] [-kb KEYBOARD] [filename]

positional arguments:
  filename              The configurator export to compile

options:
  -h, --help            show this help message and exit
  -c, --clean           Remove object files before compiling.
```

UPD: tried the same by ditching QMK and just [using miryoku branch](https://github.com/manna-harbour/miryoku_qmk/tree/miryoku) - same result.

**Comment by manna-harbour** [comment, marked as answer]
Miryoku QMK supports many keyboards, but unfortunately it can't support all keyboards automatically. The mechanism for keyboard support is described at https://github.com/manna-harbour/miryoku_qmk/tree/miryoku/users/manna-harbour_miryoku#subset-mapping. If the keyboard provides a community layout that Miryoku QMK supports then it will be supported without any extra effort. Otherwise, community layouts need to be enabled for the keyboard if a compatible layout exists, or a Miryoku subset mapping macro needs to be added for the keyboard.

Also see https://github.com/manna-harbour/miryoku/discussions/303, https://github.com/manna-harbour/miryoku/discussions/264.
